### ISSUE TITLE
Improved importer for Assignment Editor

### ISSUE THREAD

**Issue by ajtruckle** [ajtruckle/meeting-schedule-assistant-support#109]
My proposal is to replace the current [Import From OCLM Assignment History](https://www.publictalksoftware.co.uk/help-msa/srr-editor-import-mwb-history.html) feature in the **Assignment Editor**:

![image](https://github.com/user-attachments/assets/5ed8b00d-3ff7-4370-9330-e9c5317e6f01)

It will be replaced by the same [Import from Database History](https://www.publictalksoftware.co.uk/help-msa/mwb-editor-import-history.html) feature used in the **Meeting Editor**:

![image](https://github.com/user-attachments/assets/9968c35d-4ff6-4a12-94cc-64fdf6fabda7)

Some minor adjustments will need to be made to the information presented in the importer (e.g. checklist of column headings) to make it compatible with the Assignment Editor.

## Benefits ##
* Consistent GUI with both editors.
* Extra importing options (eg. CSV — Duties) will be useful.

**Comment by cengizu** [comment]
@ajtruckle 

It will be very good. A nice improvement!

**Comment by ajtruckle** [comment]
@cengizu 

## CSV — Duties ##
Existing GUI:
![image](https://github.com/user-attachments/assets/a7686910-3cac-4889-8963-99afc64ae199)

Changes required:
* Field column need to list all available duties in the CSV file. But, we must identify each duty using the Type field. Eg. Duty4P1, Duty4P2. Both have a TypeName of "Microphones" in my CSV file. So maybe, for clarity the field column should display:
    *  Duty4P1: Microphones
    *  Duty4P2: Microphones
    *  Etc.
 * Links To column will be a drop list of the visible column headings in the Assignment Editor. The list will be sorted A-Z but we will internally track the actual column position.

JSON:
The JSON mapping data will need to be something like:

```json
Type: Duty4P1
TypeName: Microphones
Include: True
LinksTo: original assignment editor column index
```

Importing:
Basically we would do this:
* Read json mapping
* Iterate CSV
    * If Type is included
        * Assign Name to LinksTo column

**Comment by ajtruckle** [comment]
@cengizu 
Looking at the **Meeting Editor** mode now. This is when we read from the _MWB history DB_. So, it is _this_ mode that will replace the original **Import from OCLM assignment history**.

Now, when we use the importer with the Meeting Editor, we allow for three columns on the mapping grid, where the middle column was for the midweek / weekend meeting. But, in the **Assignment Editor**, I do not want the complication of an additional column on the import mapping grid.

My resolution is simple. The drop-lists (Links To) cells will have the following options:

- Midweek: Host
- Midweek: Co-host
- Midweek: Zoom
- Midweek: Chairman
- Midweek: Prayer Open
- Midweek: Prayer Close
- Midweek: CBS Conductor
- Midweek: CBS Reader
- Weekend: Host
- Weekend: Co-host
- Weekend: Zoom
- Weekend: Chairman
- Weekend: Speaker
- Weekend: Theme
- Weekend: Hospitality
- Weekend: WT Conductor
- Weekend: WT Reader

These options reflect the feature we are replacing. We can also add  more options to the list, but this is a starting point. By mapping the assignment editor grid columns to the relevant options then it should work ok. There is a situation that I did not think about though, and that concerns dates / report mode.

- the MWB history groups data by week starting Monday. but it internally has the meeting dates for midweek / weekend.
- the assignment grid dates depend on the report mode.
   - so if the report mode is weekly, then we must identify the week using the Monday
   - and if the report mode is meeting, then we must use the meeting dates ti identify the data.

It is just something I have to keep in mind during importing. Either way, using the Monday will always find the right data the row(s) concerned.

Technically, importing might mess up for all modes, if the assignment editor is in Weekly mode. I might disable the importer for this report mode. We will see. 

Also, note that the assignment editor has the newer features now on the right that can identify who is used for midweek / weekend meetings. Thus, avoiding the need to actually import meeting data into columns (this was why it was originally implemented). But, I guess it is still useful to allow the user to import if desired, so that we don't lose any functionality.

**Comment by ajtruckle** [comment]
@cengizu 
### Report Mode ###
The Assignment Editor has two report modes:

* **Weekly** (one row per meeting)
* **Meeting** (two rows - one per meeting)

**Importing from OCLM history** copes with both modes because the MWB history is grouped by the "Monday".  So it is very easy to determine what data and which rows are associated with each other. The common denominator being the "Monday" being the same.

But, these import modes:

* CSV — Duties
* CSV

will only work when the report mode is **Meeting**. When in weekly mode, the "dates" displayed in the assignment editor is based on the day selected in the general options. In most cases this will be a Monday too, but it could be any day. So, using the same logic as above would require that we establish what the weekly "date" would be for each meeting (based on the program options). Then we would be able to import.

I was hoping that things would not get this complicated. But I forgot about the report mode. 😊

Summary:
* **MWB History**: Supports both reports modes
* **SRR History**: To be implemented
* **CSV Duties**: Supports Meeting report mode
* **CSV Meeting**: Not yet implemented
* **CSV**: Supports Meeting report mode

**Comment by ajtruckle** [comment]
@cengizu 
Updated help topic:

https://github.com/ajtruckle/meeting-schedule-assistant-support/wiki/srr-editor-import-history
